pytorch_lightning/trainer/training_loop.py

```diff
@@ -679,20 +681,33 @@ def run_training_batch(self, batch, batch_idx, dataloader_idx):
                     # -------------------
                     # calculate loss (train step + train step end)
                     # -------------------
+
+                    # no ddp sync at the beginning of forward or backward due to parameter changes in this or last step required
+                    no_sync = self._updated_model_last_step and isinstance(self.trainer.model, torch.nn.parallel.DistributedDataParallel)
+                    if no_sync:
+                        self.trainer.model.no_sync.__enter__()
+
                     self.training_step_and_backward(split_batch, batch_idx, opt_idx, optimizer, self.trainer.hiddens)
+
+                    if no_sync:
+                        self.trainer.model.__exit__()
+
```

Comment:
is it possible for this to logic to live in the DDP accelerators as opposed to the general training loop?